h/models/group_scope.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+import sqlalchemy as sa
+
+from h._compat import urlparse
+from h.db import Base, mixins
+
+
+class GroupScope(Base):
+    __tablename__ = 'groupscope'
+    id = sa.Column(sa.Integer, autoincrement=True, primary_key=True)
+
+    #: A fully qualified domain name, e.g. example.com, www.nytimes.com or
+    #: web.hypothes.is.
+    hostname = sa.Column('hostname', sa.UnicodeText)
+
+    groups = sa.orm.relationship(
+        'Group',
+        secondary='groupscope_group',
+        backref=sa.orm.backref('scopes', order_by='Group.name'),
+    )
```

Comment:
Many-many relationship between `Group` and `GroupScope`. Given a `Group` object, `mygroup.scopes` is a list of all its scopes. Given a `GroupScope`, `myscope.groups` is a list of all its groups.

Reply:
Sounds good.